core/src/main/scala/org/apache/spark/sql/rapids/tool/util/FSUtils.scala

```diff
@@ -73,13 +78,30 @@ object FSUtils {
 
   /**
    * Reads the content of a file as UTF-8 and closes the resources.
+   * Works with any Hadoop-compatible filesystem (local, HDFS, S3, GCS, etc.).
    */
-  def readFileContentAsUTF8(filePath: String, delim: String = "\n"): String = {
-    val source = UTF8Source.fromFile(filePath)
+  def readFileContentAsUTF8(
+      filePath: String,
+      hadoopConf: Configuration = new Configuration(),
```

Comment:
There is a utility function to get the hadoop configuration `RapidsToolsConfUtil.newHadoopConf()`
I also don't recommend initializing the hadoopConf in the argument by default. Instead, it should be defined as optional with None default value. Then inside the method, it sets it to the default `RapidsToolsConfUtil.newHadoopConf()` if it is undefined.

The difference between both ways is the laziness. In this verison, the argument will trigger config initialization too early that could affect some other logic.